src/main.cc

```diff
@@ -0,0 +1,88 @@
+#ifdef ONPC
+#include "../algo/debug.h"
+#else
+#define debug(...) 42
+#endif
+#include <cassert>
+#include <cstdint>
+#include <utility>
+#include <vector>
+#define fr first
+#define sc second
+#define pb push_back
+
+template<typename T, typename Functor>
+struct FunctorQueue {
+private:
+  std::vector<std::pair<T, T>> stack_first, stack_second;
+  Functor func;
+
+public:
+  void insert(T el);
+  void pop();
+  T get();
+
+  FunctorQueue(Functor func) : func(func) {}
+};
+
+template<typename T, typename Functor>
+void FunctorQueue<T, Functor>::insert(T el) {
+  if (stack_first.size()) {
+    stack_first.pb({el, func(el, stack_first.back().sc)});
```

Comment:
Use `.emplace_back()`
https://cengizhanvarli.medium.com/push-back-vs-emplace-back-in-c-stl-containers-2a26a099ec04